### PR TITLE
feat: Implement Folder Creation UI & Wire-iOS Integration - WPB-12012

### PR DESCRIPTION
**Pull request by agisilaos**
This PR implements the SwiftUI-based folder picker UI and its integration with Wire-iOS, building upon the previously added domain types. 

### UI Package (WireMoveToFolderUI)

- Add SwiftUI views for folder picker:
  - `CreateFolder`: Main view with navigation and empty state handling

## UI/UX
- Maintains existing create folder functionality
- Uses Wire's semantic colors and styling
- Proper error handling for folder operations

## To Test
1. Open conversation context menu
2. Select "Move to Folder"
3. Tap plus button
4. Create new folder
5. Tap Create
6. Verify folder has been created and moved to that folder
---

### Checklist

- [ ] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [ ] Description is filled and free of optional paragraphs.
- [ ] Adds/updates automated tests.

---

### UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_
- [ ] Make sure you use the API for UI elements that support large fonts.
- [ ] All colors are taken from WireDesign.ColorTheme or constructed using WireDesign.BaseColorPalette.
- [ ] New UI elements have Accessibility strings for VoiceOver.

### CODE REVIEW

File: wire-ios-sync-engine/Source/UserSession/UserSession.swift [hunk 1]
```diff
@@ -271,7 +271,7 @@ public protocol UserSession: AnyObject {
 
     func makeConversationFolderSelectionUseCase() -> UpdateConversationFolderUseCase
 
-    func makeConversationFolderCreationUseCase() -> CreateConversationFolderUseCaseProtocol
+    func makeConversationFolderCreationUseCase() -> CreateConversationFolderUseCase
```
Comment (samwyndham) [inline wire-ios-sync-engine/Source/UserSession/UserSession.swift:274]: I think this should be renamed to `makeCreateConversationFolderUseCase()`

File: wire-ios/Wire-iOS/Sources/UserInterface/Folders/Mappers/Folder+LabelType.swift [hunk 2]
```diff
@@ -17,10 +17,14 @@
...
-    /// Creates a new conversation folder with the specified name
-    /// - Parameter name: The name of the folder to be created
-    /// - Returns: The created `LabelType`, or `nil` if creation fails
-    func invoke(with name: String) async -> LabelType?
+extension Folder {
+
+    init(_ label: LabelType) {
+        self.init(
+            identifier: label.remoteIdentifier,
+            name: label.name ?? ""
```
Comment (samwyndham) [inline wire-ios/Wire-iOS/Sources/UserInterface/Folders/Mappers/Folder+LabelType.swift]: This case kind of makes no sense right where there is no name. It is kind of propagating an invalid state. Do you just use it in `CreateConversationFolderUseCase`. Consider deleting this extension and initializing with id & name. In `CreateConversationFolderUseCase` you can throw an error if the name is nil.

File: WireUI/Sources/WireMoveToFolderUI/Views/CreateFolder/CreateFolder.swift [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        conversationName: String,
+        onFolderCreated: @escaping (Folder) -> Void
+    ) {
+        self.viewModel = viewModel
+        self.conversationName = conversationName
+        self.onFolderCreated = onFolderCreated
+    }
+
+    public var body: some View {
+        NavigationStack {
```
Comment (samwyndham) [inline WireUI/Sources/WireMoveToFolderUI/Views/CreateFolder/CreateFolder.swift:43]: Personally I think the `NavigationStack` should be applied in `FolderPickerHostingController`. Doing it here limits where we can use `CreateFolder` from.

File: WireUI/Sources/WireMoveToFolderUI/Views/FolderPicker/FolderPicker.swift [hunk 4]
```diff
@@ -23,9 +23,17 @@ import WireReusableUIComponents
 public struct FolderPicker: View {
```
Comment (samwyndham) [inline WireUI/Sources/WireMoveToFolderUI/Views/FolderPicker/FolderPicker.swift:23]: I think we have both created `FolderPicker` which is a little confusing. Let's discuss naming tomorrow. I think one of us should rename our component.

**Review by samwyndham — APPROVED** [review]
Nice work!

**Review by caldrian — APPROVED** [review]

File: wire-ios-sync-engine/Source/UserSession/UserSession.swift (hunk 1 above)
Comment (agisilaos) [reply wire-ios-sync-engine/Source/UserSession/UserSession.swift:274]: Done!

File: WireUI/Sources/WireMoveToFolderUI/Views/FolderPicker/FolderPicker.swift (hunk 4 above)
Comment (agisilaos) [reply WireUI/Sources/WireMoveToFolderUI/Views/FolderPicker/FolderPicker.swift:23]: We already discussed that. Let's keep it as is for now! And we refine it together later.

File: WireUI/Sources/WireMoveToFolderUI/Views/CreateFolder/CreateFolder.swift (hunk 3 above)
Comment (agisilaos) [reply WireUI/Sources/WireMoveToFolderUI/Views/CreateFolder/CreateFolder.swift:43]: After thinking about it, I believe keeping the NavigationStack in the SwiftUI view is the better approach for this implementation. Here's why:

1. The current architecture has a clean flow:
   UIKit -> FolderPickerBuilder -> FolderPickerHostingController -> FolderPicker (SwiftUI) -> Create Folder (SwiftUI)
   
   The hosting controller serves as a bridge between UIKit and SwiftUI, while letting SwiftUI handle its own navigation concerns.

2. The SwiftUI view (FolderPicker) uses SwiftUI-specific navigation patterns:
   - NavigationLink to CreateFolder view
   - Environment values like @Environment(\.dismiss)
   - Toolbar items and navigation bar configuration
   
   Moving the NavigationStack to the hosting controller would make these harder to manage.

3. The current implementation maintains better state management where all SwiftUI-related state and navigation stays within the SwiftUI hierarchy.

File: wire-ios/Wire-iOS/Sources/UserInterface/Folders/Mappers/Folder+LabelType.swift (hunk 2 above)
Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/UserInterface/Folders/Mappers/Folder+LabelType.swift]: Simplified it by removing the extension

**Pull request merged**